lib/retriable_contexts/config.rb

```diff
@@ -0,0 +1,23 @@
+module Retriable
+  class Config
+    def self.validate_options(options)
```

Comment:
I agree with you. I think we should do strict validation. I think it might also help with api changes or major breaking changes, so that people who just upgrade will potentially see exceptions rather than unexpected or default behavior. Would you mind extracting the validation into a separate PR?